ttnn/cpp/ttnn/operations/core/work_split/work_split_tilize.hpp

```diff
@@ -24,6 +24,150 @@ struct BlockSplit {
     uint32_t nblocks_per_core_cliff;
 };
 
+struct BlockSplitWH {
+    uint32_t ncores;
+    CoreRangeSet all_cores;
+    CoreRangeSet core_range;
+    CoreRangeSet cliff_row_core_range;
+    CoreRangeSet cliff_col_core_range;
+    CoreRangeSet cliff_col_row_core_range;
+    uint32_t nblocks_per_core;
+    uint32_t single_block_size;
+    uint32_t single_block_size_cliff_row;
+    uint32_t single_block_size_cliff_col;
+    bool has_cliff_row;
+    bool has_cliff_col;
+    uint32_t full_cores_per_row;
+    uint32_t full_cores_per_col;
```

Comment:
please default-initialize all members to avoid undefined behavior

smth like

```suggestion
    uint32_t ncores = 0;
    CoreRangeSet all_cores;
    CoreRangeSet core_range;
    CoreRangeSet cliff_row_core_range;
    CoreRangeSet cliff_col_core_range;
    CoreRangeSet cliff_col_row_core_range;
    uint32_t nblocks_per_core = 0;
    uint32_t single_block_size = 0;
    uint32_t single_block_size_cliff_row = 0;
    uint32_t single_block_size_cliff_col = 0;
    bool has_cliff_row = false;
    bool has_cliff_col = false;
    uint32_t full_cores_per_row = 0;
    uint32_t full_cores_per_col = 0;
```

else you might easily get some uninitialized garbage there
https://github.com/tenstorrent/tt-metal/blob/main/contributing/BestPractices.md#15-initialize-primitive-types-on-declaration